### PR TITLE
docs: describe public api in front-page of API docs

### LINKED ISSUE #4535: we need to make clear which parts of the API docs are relevent to users, perhaps a list of user-facing classes in the API docs main page. (@ejfdickinson, edit by @martinjrobins )

### PR DESCRIPTION
**Pull request by martinjrobins**
# Description

Fixes #4535

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by valentinsulzer — CHANGES_REQUESTED** [review]
Definitely on board with having more docs like this but shouldn't it go into the user guide rather than the API page?

**Review by valentinsulzer — APPROVED** [review]
Looks good, thanks!

File: docs/source/user_guide/fundamentals/public_api.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
+----------
+Public API
+----------
+
+.. module:: pybamm
+    :noindex:
+
+PyBaMM is a Python package for mathematical modelling and simulation of battery systems. The main classes and functions that are intended to be used by the user are described in this document.
+For a more detailed description of the classes and methods, see the [API documentation](api_docs).
```
Comment (kratman) [inline docs/source/user_guide/fundamentals/public_api.rst:9]: The API documentation link does not work

**Review by kratman — CHANGES_REQUESTED** [review]
One last thing

File: docs/source/user_guide/fundamentals/public_api.rst (hunk 1 above)
Comment (martinjrobins) [reply docs/source/user_guide/fundamentals/public_api.rst:9]: whoops, thanks @kratman will fix

**Review by kratman — APPROVED** [review]

**Pull request merged**